apps/src/labs/labRedux.ts

```diff
@@ -41,6 +45,7 @@ interface LabState {
   // Whether the lab is ready for a reload.  This is used to manage the case where multiple loads
   // happen in a row, and we only want to reload the lab when we are done.
   labReadyForReload: boolean;
+  validationState: ValidationState;
```

Comment:
Wasn't sure whether this belonged here or in `progressRedux` - curious what folks think

Reply:
My initial feeling is that `progressRedux` refers to progress across levels themselves, whereas this validation system is within a single level, and that just for the sake of compartmentalizing different subsystems (and their respective states), it's fine to keep this here.

Reply:
+1, might be good to add a comment here though explaining what `validationState` is.